client/src/pages/attachments/Form.js

```diff
@@ -0,0 +1,347 @@
+import { gql } from "@apollo/client"
+import API from "api"
+import AppContext from "components/AppContext"
+import AttachmentRelatedObjectsTable from "components/Attachment/AttachmentRelatedObjectsTable"
+import ConfirmDestructive from "components/ConfirmDestructive"
+import * as FieldHelper from "components/FieldHelper"
+import Fieldset from "components/Fieldset"
+import LinkTo from "components/LinkTo"
+import Messages from "components/Messages"
+import NavigationWarning from "components/NavigationWarning"
+import { jumpToTop } from "components/Page"
+import RichTextEditor from "components/RichTextEditor"
+import { FastField, Field, Form, Formik } from "formik"
+import _isEqual from "lodash/isEqual"
+import { Attachment } from "models"
+import PropTypes from "prop-types"
+import React, { useContext, useState } from "react"
+import { Button, Col } from "react-bootstrap"
+import { useNavigate, useParams } from "react-router-dom"
+import Settings from "settings"
+import pdf from "resources/newPDF.svg"
+
+const GQL_CREATE_ATTACHMENT = gql`
+  mutation ($attachment: AttachmentInput!) {
+    createAttachment(attachment: $attachment)
+  }
+`
+
+const GQL_UPDATE_ATTACHMENT = gql`
+  mutation ($attachment: AttachmentInput!) {
+    updateAttachment(attachment: $attachment)
+  }
+`
+
+const GQL_DELETE_ATTACHMENT = gql`
+  mutation ($uuid: String!) {
+    deleteAttachment(uuid: $uuid)
+  }
+`
+
+const GQL_GET_ATTACHMENT = gql`
+  query ($uuid: String) {
+    attachment(uuid: $uuid) {
+      content
+      mimeType
+      attachmentRelatedObjects {
+        relatedObject {
+          ... on AuthorizationGroup {
+            name
+          }
+          ... on Location {
+            name
+          }
+          ... on Organization {
+            shortName
+          }
+          ... on Person {
+            role
+            rank
+            name
+            avatar(size: 32)
+          }
+          ... on Position {
+            type
+            name
+          }
+          ... on Report {
+            intent
+            engagementDate
+            state
+          }
+          ... on Task {
+            shortName
+            longName
+          }
+        }
+        relatedObjectUuid
+        relatedObjectType
+      }
+    }
+  }
+`
+
+const AttachmentForm = ({ edit, title, initialValues }) => {
+  const navigate = useNavigate()
+  const { uuid } = useParams()
+  const { currentUser } = useContext(AppContext)
+  const { data } = API.useApiQuery(GQL_GET_ATTACHMENT, { uuid })
+  const [error, setError] = useState(null)
+  const canEdit =
+    currentUser.isAdmin() || currentUser.uuid === initialValues.author.uuid
+
+  const classificationButtons = [
+    {
+      id: Settings.fields.attachment.classification.choices.UNDEFINED.label,
```

Comment:
Why not use `Settings.fields.attachment.classification.choices.map(…)` to create this array?